### PR TITLE
[16.0][IMP] helpdesk_mgmt_sale: add button create ticket and display sales

### PR DESCRIPTION
**Pull request by matthieusaison**
add button to create a ticket with value of active sale order.
display and default filled sale_order_ids.

### CODE REVIEW

**Comment by CristianoMafraJunior** [comment]
@matthieusaison Can you increase the test coverage?

File: helpdesk_mgmt_sale/views/sale_order_views.xml [hunk 1]
```diff
@@ -28,6 +28,14 @@
                     <field string="Tickets" name="ticket_count" widget="statinfo" />
                 </button>
             </xpath>
+            <xpath expr="//header" position="inside">
+                <button
+                    name="action_create_helpdesk_ticket"
+                    type="object"
+                    string="Créer un ticket"
```
Comment (luisDIXMIT) [inline helpdesk_mgmt_sale/views/sale_order_views.xml]: Text must be in english and add a translation to your language.

File: helpdesk_mgmt_sale/views/helpdesk_ticket_views.xml [hunk 2]
```diff
@@ -17,6 +17,9 @@
                     <field string="Sale Orders" name="so_count" widget="statinfo" />
                 </button>
             </xpath>
+            <field name="channel_id" position="after">
+                <field name="sale_order_ids" widget="many2many_tags" />
```
Comment (luisDIXMIT) [inline helpdesk_mgmt_sale/views/helpdesk_ticket_views.xml:21]: If sale orders are already displayed in the smart button, what is the need to add it here?

File: helpdesk_mgmt_sale/views/sale_order_views.xml (hunk 1 above)
Comment (matthieusaison) [reply helpdesk_mgmt_sale/views/sale_order_views.xml]: Yes, sorry it's a mistake, I fix this in my last commit

File: helpdesk_mgmt_sale/views/helpdesk_ticket_views.xml (hunk 2 above)
Comment (matthieusaison) [reply helpdesk_mgmt_sale/views/helpdesk_ticket_views.xml:21]: When you create a ticket from sale, you may need to add an other sale linked. And if you create ticket from helpdesk menu, you may need to link to sale directly.

Comment (luisDIXMIT) [reply helpdesk_mgmt_sale/views/helpdesk_ticket_views.xml:21]: Ok, I see your point of view. It seems strange to me to see duplicate data about sale orders in both the field and the button. However, it is true that from the button it is not possible to add existing sale orders, only to create a new one. Anyway, LGTM!

**Review by luisDIXMIT — APPROVED** [review]
LGTM!

**Review by pedrobaeza — CHANGES_REQUESTED** [review]
Sorry, I don't think this is the way, but to pass in the existing smart-button the `default_...` context key for using it when pressing "Create" button inside ticket list.

**Review by marcos-mendez — CHANGES_REQUESTED** [review]
**Automated Review -- Tests Failed**

### 1. **Root Cause**
The test failure is due to a **database connection error** (`Connection to the database failed`) during the Odoo server startup, not directly caused by the code changes in the PR. This is likely a transient infrastructure issue in the test environment (e.g., Runboat).

---

### 2. **Suggested Fix**
No functional code fix is needed for the test failure itself, but to **prevent future failures**, ensure that:
- The test environment has a stable database connection.
- The Odoo server is started with proper database configuration.

However, in terms of code correctness:
- The `create` method in `helpdesk.ticket` correctly uses `self.env.context.get("from_sale_order")` and is safe.
- The `action_create_helpdesk_ticket` method in `sale.order` correctly sets the context with `from_sale_order: True`.

---

### 3. **Additional Code Issues**
- **Missing `@api.model` decorator**: In `helpdesk.ticket.create`, the method `create` is correctly defined with `@api.model_create_multi`, which is fine. However, the method **should not be using `self`** inside the loop as `self` refers to the class, not the instance. This is a **potential bug**.

  **Fix**:
  ```python
  # Replace:
  for sale in tickets.sale_order_ids:
      sale.message_post(...)
  # With:
  for ticket in tickets:
      for sale in ticket.sale_order_ids:
          sale.message_post(...)
  ```

  **File**: `helpdesk_mgmt_sale/models/helpdesk_ticket.py`  
  **Lines**: 29–36

---

### 4. **Test Improvements**
The existing tests are good, but can be improved to better cover edge cases and ensure robustness:

#### Suggested Tests:
1. **Test with multiple sale orders**:
   - Create a ticket with multiple `sale_order_ids`.
   - Ensure messages are posted for each sale order.

2. **Test with empty sale orders list**:
   - Create a ticket without any `sale_order_ids`.
   - Ensure no message is posted.

3. **Test `action_create_helpdesk_ticket` with different user contexts**:
   - Use `with_user()` to test behavior with different users.
   - Ensure correct `default_partner_id` and `default_name` are set.

4. **Use `SavepointCase` for transaction safety**:
   - The current `BaseCommon` (likely `TransactionCase`) is acceptable, but for better test isolation, consider using `SavepointCase` for tests involving `message_post`.

#### OCA Testing Patterns:
- Use `SavepointCase` for tests involving `message_post` or complex data changes.
- Use `tagged` to mark tests for specific scenarios (e.g., `@tagged('post_install')` for context-based logic).

---

### Summary

| Item | Status |
|------|--------|
| Root Cause | Database connection issue (environmental) |
| Code Bug | `self` used instead of `ticket` in loop (line 30) |
| Test Coverage | Good, but can be improved with edge cases |
| Suggested Fix | Replace `self` with `ticket` in message posting loop |

Let me know if you'd like a patch for the bug fix.

---
### Reciprocal Review Request



Hi everyone! I found some test failures on this PR and left detailed feedback above. I am happy to discuss or help debug. In the meantime, if any of you get a chance, I would appreciate a look at my open PR(s):

**My open PRs across OCA:**
- [server-tools#3554 [MIG] datetime_formatter: Migration to 18.0](https://github.com/OCA/server-tools/pull/3554)
- [server-tools#3548 [18.0][MIG] base_kanban_stage: Migration to 18.0](https://github.com/OCA/server-tools/pull/3548)
- [hr-attendance#262 [16.0][ADD] Hr_attendance_idsecure: iDSecure (ControliD) attendance integration](https://github.com/OCA/hr-attendance/pull/262)
- [stock-logistics-workflow#2276 [16.0][ADD] stock_move_line_devaluation](https://github.com/OCA/stock-logistics-workflow/pull/2276)
- [stock-logistics-workflow#2275 [16.0][ADD] Stock move line analytic account](https://github.com/OCA/stock-logistics-workflow/pull/2275)
- [stock-logistics-workflow#2268 [16.0][ADD] stock_move_line_picking_partner](https://github.com/OCA/stock-logistics-workflow/pull/2268)
- [purchase-workflow#2694 [16.0][IMP]Purchase workflow added to review state & exception fix](https://github.com/OCA/purchase-workflow/pull/2694)

Reviewing each other's work helps the whole community move forward. Thank you!

---
_Environment via [OCA Neural Reviewer](https://git.pop.coop/pop/oca-reviewer): Minikube + K8s Job + oca-ci/py3.10-odoo16.0 | Odoo 16.0_
_Automated review by [OCA Neural Reviewer](https://git.pop.coop/pop/oca-reviewer) + qwen3-coder:30b_
